### PR TITLE
[feature] Integrate Ink Dialogue Tools

### LINKED ISSUE #177: Dialog box with Inky / JSON parser

### PR DESCRIPTION
**Pull request by uciii**
### Main changes
Integrate Ink to provide JSON to our dialogue flow

### Tasks List
- [close #177]

### Attachments
1. Ink folder (consist JSON file)
image

2. Edit json in inky tools
image

3. Dialogue Box prefab > DialogueManager.cs > variable Dialogue Json should refer to Ink Dialogue.json
image

4. Result
image

### New Packages
- [Ink Unity Integration](https://assetstore.unity.com/packages/tools/integration/ink-unity-integration-60055)
- Extra : [Inky tools](https://www.inklestudios.com/ink) (Ink editor)

### CODE REVIEW

File: Assets/Astralization/Scripts/UI/Ink/InkScript.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
+using Ink.Runtime;
```
Comment (raihansyahdean) [inline Assets/Astralization/Scripts/UI/Ink/InkScript.cs:1]: remove this class

File: Assets/Astralization/Scripts/UI/DialogueManager.cs [hunk 2]
```diff
@@ -24,10 +25,11 @@ public class DialogueManager : MonoBehaviour, IDialogueManager
...
     private Text _dialogueText;
     [SerializeField]
-    private Animator _animator;
-
-    public string[] lines;
-    private int _index;
+    private Animator _animator;
+
+    [SerializeField]
+    private TextAsset _dialogueJson;
```
Comment (raihansyahdean) [inline Assets/Astralization/Scripts/UI/DialogueManager.cs:31]: make setter for _dialogueJson

**Review by raihansyahdean — CHANGES_REQUESTED** [review]
Move resource files to Astralization/Resources/Dialogue/DummyText.json etc.

**Review by VadimBhirawa — APPROVED** [review]
LGTM

File: Assets/Astralization/Scripts/UI/DialogueManager.cs [hunk 3]
```diff
@@ -2,13 +2,14 @@
 using UnityEngine.UI;
 using System;
 using System.Collections;
-
+using Ink.Runtime;
+
 public interface IDialogueManager
 {
     Animator GetAnimator();
     bool GetDialogBox();
```
Comment (rafiftaris) [inline Assets/Astralization/Scripts/UI/DialogueManager.cs:10]: error CS0535: 'DialogueManager' does not implement interface member 'IDialogueManager.GetDialogBox()'

Rename this

**Review by rafiftaris — CHANGES_REQUESTED** [review]
perlu docs?

**Review by rafiftaris — APPROVED** [review]

**Pull request merged**